File: 733-flood-fill/733-flood-fill.py
class Solution:
    
    def Traverse(self, img, sr, sc, origin_color, newColor):
        
        if(sr<0 or sr>=len(img) or sc<0 or sc>=len(img[0]) or img[sr][sc]!=origin_color or img[sr][sc]==newColor):
            return 0
        
        img[sr][sc]=newColor
        
        self.Traverse(img, sr+1, sc, origin_color, newColor)
        self.Traverse(img, sr-1, sc, origin_color, newColor)
        self.Traverse(img, sr, sc+1, origin_color, newColor)
        self.Traverse(img, sr, sc-1, origin_color, newColor)
    
    
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        
        
        self.Traverse(image, sr, sc, image[sr][sc], newColor)
        
        return image

File: 733-flood-fill/733-flood-fill.cpp
class Solution {
public:
    
    void Traverse(vector<vector<int>>& img, int sr, int sc, int origin_color, int newColor){
        
        if(sr<0 || sc<0 || sr>=img.size() || sc>=img[0].size() || img[sr][sc]==newColor || img[sr][sc]!=origin_color)
            return;
        
        img[sr][sc]=newColor;
        
        Traverse(img, sr+1, sc, origin_color, newColor);
        Traverse(img, sr-1, sc, origin_color, newColor);
        Traverse(img, sr, sc+1, origin_color, newColor);
        Traverse(img, sr, sc-1, origin_color, newColor);        
    }
    
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int newColor) {
        
        Traverse(image, sr, sc, image[sr][sc], newColor);
        
        return image;
    }
};